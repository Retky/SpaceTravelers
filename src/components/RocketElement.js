import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { reserveRocket, unreserveRocket } from '../redux/rockets/Rockets';

const RocketElement = (props) => {
  const dispatch = useDispatch();
  const {
    id, name, description, flickrImages, reserved,
  } = props;

  RocketElement.defaultProps = {
    reserved: false,
  };

  const reserveRocketHandler = (e) => {
    if (reserved) {
      dispatch(unreserveRocket(e.target.id));
    } else {
      dispatch(reserveRocket(e.target.id));
    }
  };
  const reservation = reserved ? 'Cancel Reservation' : 'Reserve Rocket';

  return (
    <div className="rocketEl">
      <div className="rocketImg">
        <img src={flickrImages} alt={description} />
      </div>
      <div>
        <h2>{name}</h2>
        <p>
          {(reserved) && (<span className="badge">Reserved</span>)}
          {description}
        </p>
        <button type="button" id={id} onClick={reserveRocketHandler}>{reservation}</button>
      </div>
    </div>
  );
};

RocketElement.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  reserved: PropTypes.bool,
  description: PropTypes.string.isRequired,
  flickrImages: PropTypes.string.isRequired,
};

export default RocketElement;
