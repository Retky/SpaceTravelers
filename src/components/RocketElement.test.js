import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import RocketElement from './RocketElement';

const rocket = {
  rocket_id: '654',
  rocket_name: 'Rocket',
  description: 'Lorem',
  flickr_images: ['URL'],
  reserved: false,
};

describe('RocketElement component', () => {
  test('component match with the snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <RocketElement
          key={rocket.rocket_id}
          id={rocket.rocket_id}
          name={rocket.rocket_name}
          description={rocket.description}
          flickrImages={rocket.flickr_images[0]}
          reserved={rocket.reserved}
        />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
