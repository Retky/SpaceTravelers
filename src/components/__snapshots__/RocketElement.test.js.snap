// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketElement component component match with the snapshot 1`] = `
<div
  className="rocketEl"
>
  <div
    className="rocketImg"
  >
    <img
      alt="Lorem"
      src="URL"
    />
  </div>
  <div>
    <h2>
      Rocket
    </h2>
    <p>
      Lorem
    </p>
    <button
      className="blueBtn"
      id="654"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
